.github/workflows/nightly_vtk_master.yml

```
@@ -132,15 +133,44 @@ jobs:
         lfs_sha: ${{ needs.cache_lfs.outputs.lfs_sha}}
 
 #----------------------------------------------------------------------------
-# MacOS CI: Build and test, cross-cpu build matrix
+# MacOS CI: Build and test
 #----------------------------------------------------------------------------
   macos:
     needs: [cache_lfs, check_nightly]
     if: ${{ needs.check_nightly.outputs.should_run == 'true' }}
     strategy:
       fail-fast: false
       matrix:
-        cpu: [x86_64, arm64]
+        cpu: [x86_64]
+
+    runs-on: macos-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        path: 'source'
+        fetch-depth: 0
+        lfs: false
+
+    - name: Generic CI
+      uses: ./source/.github/actions/generic-ci
+      with:
+        build_type: ${{matrix.build_type}}
+        vtk_version: ${{needs.check_nightly.outputs.vtk_sha}}
+        cpu: ${{matrix.cpu}}
+        lfs_sha: ${{ needs.cache_lfs.outputs.lfs_sha}}
+
+#----------------------------------------------------------------------------
+# MacOS arm64 CI: Build and test
+#----------------------------------------------------------------------------
+  macos_arm:
+    needs: [cache_lfs, check_nightly]
+    if: ${{ needs.check_nightly.outputs.should_run == 'true' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cpu: [arm64]
 
     runs-on: macos-latest
```

Should it run on `macos-14`?

```suggestion
    runs-on: macos-14
```